escrow/free_call_storage.go

```diff
@@ -0,0 +1,55 @@
+package escrow
+
+import (
+	"github.com/singnet/snet-daemon/blockchain"
+	"reflect"
+)
+
+type FreeCallUserStorage struct {
+	delegate TypedAtomicStorage
+}
+
+func NewFreeCallUserStorage(atomicStorage AtomicStorage, metadata *blockchain.ServiceMetadata) *FreeCallUserStorage {
+	return &FreeCallUserStorage{
+		delegate: &TypedAtomicStorageImpl{
+			atomicStorage: &PrefixedAtomicStorage{
+				delegate: atomicStorage,
+				//Add the MPE Network address as the prefix on the key for storage
+				keyPrefix: "/" + metadata.MpeAddress + "/free-call-user/storage",
```

Comment:
Prefixing number of free calls used by MPE contract address means that after MPE contract update all users will have zero number of free calls made. I suggest removing MpeAddress prefix here.